vhost-user-backend/src/backend.rs

```diff
@@ -87,13 +86,17 @@ pub trait VhostUserBackend: Send + Sync {
     /// function.
     fn set_backend_req_fd(&self, _backend: Backend) {}
 
-    #[cfg(feature = "gpu-socket")]
     /// Set handler for communicating with the frontend by the gpu specific backend communication
     /// channel.
     ///
-    /// This method only exits when the crate feature gpu-socket is enabled, because this is only
-    /// useful for a gpu device.
-    fn set_gpu_socket(&self, _gpu_backend: GpuBackend);
+    /// This function returns a `Result`, returning an error if the backend does not implement this
+    /// function.
+    fn set_gpu_socket(&self, _gpu_backend: GpuBackend) -> Result<()> {
+        Err(std::io::Error::new(
+            std::io::ErrorKind::Unsupported,
+            "back end does not support set_gpu_socket() / VHOST_USER_GPU_SET_SOCKET",
```

Comment:
```suggestion
            "backend does not support set_gpu_socket() / VHOST_USER_GPU_SET_SOCKET",
```

Reply:
In `vhost-user-backend/src/backend.rs` all other error messages already use `back end`. I don't know why, and I also prefer `backend`, but for consistently I'd suggest staying with `back end` for now. I can send another PR later to fix all of them. WDYT?

Reply:
> In `vhost-user-backend/src/backend.rs` all other error messages already use `back end`. I don't know why, and I also prefer `backend`, but for consistently I'd suggest staying with `back end` for now. I can send another PR later to fix all of them. WDYT?

🫨 I can help there, if you are willing to delegate to me

Reply:
> In `vhost-user-backend/src/backend.rs` all other error messages already use `back end`. I don't know why, and I also prefer `backend`, but for consistently I'd suggest staying with `back end` for now. I can send another PR later to fix all of them. WDYT?

I agree, personally I don't like adding unorthodox spellings for consistency even if they are to be fixed later. I realise this is really unimportant and I'm holding the PR up for this 🥺 I thought it'd be easy to fix by dorinda though (I'd do it myself but I can't push at her repo).

It's not fixed yet so I will change my review to approval and not changes requested. I wouldn't want to inconvenience for minutiae!

Reply:
> I agree, personally I don't like adding unorthodox spellings for consistency even if they are to be fixed later.

Yep, this is a good point.

>I realise this is really unimportant and I'm holding the PR up for this 🥺 I thought it'd be easy to fix by dorinda though (I'd do it myself but I can't push at her repo).

@dorindabassey up to you if you want to fix it now, or we can fix later, but I agree that we can replace "back end" with "backend". @TimePrinciple if you have time, feel free to check that, maybe also for the frontend.

Reply:
@stefano-garzarella I don't mind, I can fix it in this PR.